lib/app/views/application/_nav.erb

```diff
@@ -2,6 +2,22 @@
   <ul class="nav navbar-nav">
     <li><a href="http://help.exercism.io">Help</a></li>
     <li><a href="/getting-started">Welcome new users</a></li>
+    <li class="dropdown">
+      <a class="dropdown-toggle" id="drop-current" href="#" data-toggle="dropdown">Languages<b class="caret"></b></a>
+    <ul class="dropdown-menu" id="menu-current" role="menu" area-labelledby="drop-current">
+          <% all_languages.each do |l| %>
```

Comment:
Interesting observations. I have a preference for symmetry, so my favorite is the third one, where each is clearly the singularization/pluralization of the other.

In Ruby it's very uncommon to see single-letter variable names (except `i` for looping indices sometimes, or for a small int, and `s` for a generic string in a very small scope).

I would suggest using `tracks.each do |track|` for this section.